pkg/cmd/fixcerts/fixcerts.go

```diff
@@ -0,0 +1,431 @@
+	certRotationWg.Add(1)
+	go func() {
+		defer certRotationWg.Done()
+		configInformers.Start(certRotationCtx.Done())
+	}()
+
+	kubeApiserverOperatorClient := &kubeapiserveroperatorclient.OperatorClient{
+		Informers: operatorConfigInformers,
+		Client:    operatorConfigClient.OperatorV1(),
+	}
+
+	kubeControllerManagerOperatorClient := &kubecontrollermanageroperatorclient.OperatorClient{
+		Informers: operatorConfigInformers,
+		Client:    operatorConfigClient.OperatorV1(),
+	}
+
+	kubeApiserverInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(
+		kubeClient,
+		"",
+		kubeapiserveroperatorclient.GlobalUserSpecifiedConfigNamespace,
+		kubeapiserveroperatorclient.GlobalMachineSpecifiedConfigNamespace,
+		kubeapiserveroperatorclient.TargetNamespace,
+		kubeapiserveroperatorclient.OperatorNamespace,
+		"kube-system",
+	)
+	certRotationWg.Add(1)
+	go func() {
+		defer certRotationWg.Done()
+		kubeApiserverInformersForNamespaces.Start(certRotationCtx.Done())
+	}()
+
+	kubeControllerManagerInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(
+		kubeClient,
+		"",
+		kubecontrollermanageroperatorclient.GlobalUserSpecifiedConfigNamespace,
+		kubecontrollermanageroperatorclient.GlobalMachineSpecifiedConfigNamespace,
+		kubecontrollermanageroperatorclient.TargetNamespace,
+		kubecontrollermanageroperatorclient.OperatorNamespace,
+		"kube-system",
+	)
+	certRotationWg.Add(1)
+	go func() {
+		defer certRotationWg.Done()
+		kubeControllerManagerInformersForNamespaces.Start(certRotationCtx.Done())
+	}()
+
+	eventRecorder := events.NewKubeRecorder(kubeClient.CoreV1().Events(""), "fix-certs (CLI)", &corev1.ObjectReference{
+		APIVersion: "v1",
+		Kind:       "namespace",
+		Name:       "kube-system",
+	}) // fake
+
+	kubeApiserverCertRotationController, err := kubeapiservercertrotationcontroller.NewCertRotationController(
+		kubeClient,
+		kubeApiserverOperatorClient,
+		configInformers,
+		kubeApiserverInformersForNamespaces,
+		eventRecorder.WithComponentSuffix("cert-rotation-controller-kas"),
+		0,
+	)
+	if err != nil {
+		return err
+	}
+	certRotationWg.Add(1)
+	go func() {
+		defer certRotationWg.Done()
+
+		kubeApiserverCertRotationController.Run(1, certRotationCtx.Done())
+	}()
+
+	kubeControllerManagerCertRotationController, err := kubecontrollermanagercertrotationcontroller.NewCertRotationController(
+		kubeClient.CoreV1(),
+		kubeClient.CoreV1(),
+		kubeControllerManagerOperatorClient,
+		kubeControllerManagerInformersForNamespaces,
+		eventRecorder.WithComponentSuffix("cert-rotation-controller-kcm"),
+		0,
+	)
+	if err != nil {
+		return err
+	}
+	certRotationWg.Add(1)
+	go func() {
+		defer certRotationWg.Done()
+
+		kubeControllerManagerCertRotationController.Run(1, certRotationCtx.Done())
+	}()
+
+	klog.Info("Waiting for certs to be refreshed...")
+	// FIXME: wait for valid certs
+	// time.Sleep(5*time.Minute)
+	time.Sleep(30 * time.Second)
```

Comment:
This won't do.  Add a `RunOnce` at each level so we can be sure we run and avoid the go func/defer, wait, hope flow.

Reply:
> This won't do. Add a `RunOnce` at each level so we can be sure we run and avoid the go func/defer, wait, hope flow.

You'll also need a `WaitForPrereqs` func to sync the bits at least once.